pkg/templates/accesstoresources/template.go

```diff
@@ -0,0 +1,206 @@
+			}
+			if r.AggregationRule != nil && len(r.AggregationRule.ClusterRoleSelectors) > 0 {
+				resultsAggregated := findAggregatedAccesses(clusterroles, r.AggregationRule.ClusterRoleSelectors, resourceRegexes, verbRegexes)
+				results = append(results, resultsAggregated...)
+			}
+		}
+	}
+	if !roleExists && flag {
+		results = append(results, diagnostic.Diagnostic{Message: fmt.Sprintf("clusterrole %q not found", name)})
+	}
+	return results
+}
+
+// find clusterroles by label selectors, and check if they have access to the specified resources and verbs
+func findAggregatedAccesses(clusterroles []*rbacV1.ClusterRole, selectors []metaV1.LabelSelector, resourceRegexes, verbRegexes []*regexp.Regexp) []diagnostic.Diagnostic {
+	results := []diagnostic.Diagnostic{}
+	for _, s := range selectors {
+		labelSelector, err := metaV1.LabelSelectorAsSelector(&metaV1.LabelSelector{MatchLabels: s.MatchLabels})
+		if err != nil {
+			continue
+		}
+		for _, r := range clusterroles {
+			if labelSelector.Matches(labels.Set(r.GetLabels())) { // Found the aggregated clusterrole!
+				accesses := checkAccess(r.Rules, resourceRegexes, verbRegexes)
+				if len(accesses) > 0 {
+					results = append(results, diagnostic.Diagnostic{Message: fmt.Sprintf("binding via aggregationRule to %q clusterrole that has %s", r.Name, strings.Join(accesses, ", "))})
+				}
+			}
+		}
+	}
+	return results
+}
+
+// find role by name and namespace that has access to the specified resources and verbs
+func findRole(name, namespace string, lintCtx lintcontext.LintContext, resources, verbs []*regexp.Regexp, flag bool) []diagnostic.Diagnostic {
+	results := []diagnostic.Diagnostic{}
+	roleExists := false
+	for _, object := range lintCtx.Objects() {
+		gvk := extract.GVK(object.K8sObject)
+		if gvk != roleGVK {
+			continue
+		}
+		r, ok := object.K8sObject.(*rbacV1.Role)
+		if !ok {
+			continue
+		}
+		ns := stringutils.OrDefault(r.Namespace, "default")
+		if r.Name == name && ns == namespace {
+			roleExists = true
+			accesses := checkAccess(r.Rules, resources, verbs)
+			if len(accesses) > 0 {
+				results = append(results, diagnostic.Diagnostic{Message: fmt.Sprintf("binding to %q role that has %s", r.Name, strings.Join(accesses, ", "))})
+			}
+		}
+	}
+	if !roleExists && flag {
+		results = append(results, diagnostic.Diagnostic{Message: fmt.Sprintf("role %q in namespace %q not found", name, namespace)})
+	}
+	return results
+}
+
+// find access verbs to a given resource kind
+func checkAccess(rules []rbacV1.PolicyRule, resourceRegex, verbRegex []*regexp.Regexp) []string {
+	var accesses, resources, verbs []string
+	for _, rule := range rules {
+		resources = []string{}
+		for _, res := range rule.Resources {
+			if isInList(resourceRegex, res) || res == "*" {
+				resources = append(resources, res)
+			}
+		}
+		if len(resources) > 0 {
+			verbs = []string{}
+			for _, verb := range rule.Verbs {
+				if isInList(verbRegex, verb) || verb == "*" {
```

Comment:
Reverse the order

Reply:
removed res == "*" from the if condition because isInList(resourceRegex, res) handles it already.